python/pyhail/dataset.py

```diff
@@ -177,12 +177,58 @@ def annotate_global_list(self, input, root, as_set=False):
         """Load text file into global annotations as Array[String] or
         Set[String].
 
+        **Examples**
+
+        Let's add a gene list into global annotations::
+
+          $ cat data/genes.txt
+          SCN2A
+          SONIC-HEDGEHOG
+          PRNP
+          ALDH4A1
+          LEP
+          OSM
+          TSC1
+          TSC2
+
+        To annotate with this list as type ``Array[String]``:
+
+        >>> vds = (hc.read('data/example.vds')
+        >>>  .annotate_global_list('data/genes.txt', 'global.genes'))
+
+        To see the resulting global annotation schema and values:
+
+        >>> vds.print_schema(print_global=True)
+        >>>  .show_globals()
```

Comment:
There's a question about scoping here.  Are variables set in one code block visible in the next?  Is the scoping per module, class, function, or code block?

My guess from a quick look at doctest:

http://www.sphinx-doc.org/en/1.5.1/ext/doctest.html

is scoping is per code block.  Let's require that for now.